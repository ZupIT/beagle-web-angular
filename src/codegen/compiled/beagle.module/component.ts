/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { Type } from '@angular/core'
import {
  kebabToCamelCase,
  removeExtraIndentation,
  replaceToUnderline,
} from '../../utils/formatting'
import { getComponentAnnotations } from '../../utils/metadata'
import { remoteViewSelector } from '../../../constants'


function createViewChildString(name: string, templateName: string, angularVersion: number): string {
  const componentName = replaceToUnderline(name)

  if (angularVersion >= 8) {
    return `'${componentName}': new ViewChild('${templateName}', { static: true })`
  } else {
    return `'${componentName}': new ViewChild('${templateName}')`
  }
}

function createQueries(components: Record<string, Type<any>>, angularVersion: number) {
  const componentNames = Object.keys(components)

  return componentNames.map((name) => {
    const selector = getComponentAnnotations(components[name]).selector
    const templateName = kebabToCamelCase(selector)
    return createViewChildString(name, templateName, angularVersion)
  })
}


export function createComponentString(
    components: Record<string, Type<any>>,
    angularVersion: number,
  ) {
  const queries = createQueries(components, angularVersion)

  // todo: legacy code. Remove the input "loadParams" with v2.0.
  const componentString = `
    @Component({
      selector: '${remoteViewSelector}',
      template,
      inputs: ['loadParams', 'route', 'networkOptions', 'controllerId'],
      queries: {
        ${queries.join(',\n        ')},
      },
    })
    export class BeagleRemoteView extends AbstractBeagleRemoteView {
      constructor(
        beagleProvider: BeagleProvider,
        ngZone: NgZone,
        changeDetector: ChangeDetectorRef,
      ) {
        // @ts-ignore
        super(beagleProvider, ngZone, changeDetector)
      }
    }
  `

  return removeExtraIndentation(componentString, 4)
}
