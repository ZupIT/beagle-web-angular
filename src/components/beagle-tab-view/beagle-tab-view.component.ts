/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import {
  Component, AfterViewInit, ViewChild, 
  Input, OnInit, ViewEncapsulation,
} from '@angular/core'
import { TabsService } from '../services/tabs.service'

@Component({
  selector: 'beagle-tab-view',
  templateUrl: './beagle-tab-view.component.html',
  styleUrls: ['./beagle-tab-view.component.less'],
  encapsulation: ViewEncapsulation.None,
})
export class BeagleTabViewComponent implements AfterViewInit, OnInit {
  @Input() styleId?= ''
  @ViewChild('contentItens') contentItens
  activeTab = ''

  constructor(
    private tabsService: TabsService) {
  }

  ngOnInit() {
    this.listenTabChanges()
  }

  ngAfterViewInit() {
    if (this.contentItens) this.initializeTabSelected()
  }

  listenTabChanges() {
    this.tabsService.notifySelectedTab().subscribe((selected) => {
      this.initializeTabSelected()
    })
  }

  initializeTabSelected() {
    if (this.contentItens) {
      const elements: Element[] = Array.from(this.contentItens.nativeElement.children)
      elements.forEach((item, index) => {
        const viewId = item.getAttribute('ng-reflect-_element-id')
        if (this.activeTab === '' && index === 0) {
          this.activeTab = viewId || ''
          this.tabsService.changeSelectedTab(this.activeTab)
        }
      })
    }
  }
}
