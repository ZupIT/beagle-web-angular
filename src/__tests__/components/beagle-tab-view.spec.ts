/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { TestBed, async } from '@angular/core/testing'
import { BeagleTabViewComponent } from '../../components/beagle-tab-view/beagle-tab-view.component'
import { TabsService } from '../../components/services/tabs.service'

let component: BeagleTabViewComponent

describe('BeagleTabViewComponent', () => {


  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        BeagleTabViewComponent,
      ],
      providers: [
        TabsService,
      ],
    }).compileComponents()
    const fixture = TestBed.createComponent(BeagleTabViewComponent)
    component = fixture.componentInstance
  }))

  it('should match the snapshot', () => {
    expect(component).toMatchSnapshot()
  })


})
