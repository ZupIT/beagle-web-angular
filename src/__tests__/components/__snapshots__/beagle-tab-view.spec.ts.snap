// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeagleTabViewComponent should match the snapshot 1`] = `
BeagleTabViewComponent {
  "activeTab": "",
  "styleId": "",
  "tabsService": TabsService {
    "selectedTab": BehaviorSubject {
      "_isScalar": false,
      "_value": "",
      "closed": false,
      "hasError": false,
      "isStopped": false,
      "observers": Array [],
      "thrownError": null,
    },
  },
}
`;
